File: server/app/schemas/embedding_config.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class EmbeddingConfigBase(BaseModel):
    name: str
    model_id: str
    base_url: str
    api_key: str
    description: str = ''


class EmbeddingConfigCreate(EmbeddingConfigBase):
    pass


class EmbeddingConfigUpdate(BaseModel):
    name: Optional[str] = None
    model_id: Optional[str] = None
    base_url: Optional[str] = None
    api_key: Optional[str] = None
    description: Optional[str] = None


class EmbeddingConfigResponse(EmbeddingConfigBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: server_go/internal/schema/embedding_config.go
package schema

import "time"

type EmbeddingConfigBase struct {
	Name        string `json:"name" binding:"required"`
	ModelID     string `json:"model_id" binding:"required"`
	BaseURL     string `json:"base_url" binding:"required"`
	APIKey      string `json:"api_key" binding:"required"`
	Description string `json:"description"`
}

type EmbeddingConfigCreate EmbeddingConfigBase

type EmbeddingConfigUpdate struct {
	Name        *string `json:"name"`
	ModelID     *string `json:"model_id"`
	BaseURL     *string `json:"base_url"`
	APIKey      *string `json:"api_key"`
	Description *string `json:"description"`
}

type EmbeddingConfigResponse struct {
	ID          int64     `json:"id"`
	Name        string    `json:"name"`
	ModelID     string    `json:"model_id"`
	BaseURL     string    `json:"base_url"`
	APIKey      string    `json:"api_key"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
